hkmc2/shared/src/test/mlscript/objbuf/Mutation.mls

```diff
@@ -0,0 +1,150 @@
+:js
+
+import "../../mlscript-compile/ObjectBuffer.mls"
+open ObjectBuffer
+open annotations
+
+
+:sjs
+@bufferable
+class A(x) with
+  fun f(y) =
+    set x += 1
+    x
+//│ JS (unsanitized):
+//│ let A1;
+//│ A1 = function A(x) {
+//│   return globalThis.Object.freeze(new A.class(x));
+//│ };
+//│ globalThis.Object.freeze(class A {
+//│   static {
+//│     A1.class = this
+//│   }
+//│   constructor(x) {
+//│     this.#x = x;
+//│   }
+//│   static {
+//│     this.size = 1;
+//│   }
+//│   static ctor(buf, idx) {
+//│     return (x) => {
+//│       let idx1;
+//│       idx1 = idx + 0;
+//│       buf.buf[idx1] = x;
+//│       return idx
+//│     }
+//│   } 
+//│   static f(buf, idx) {
+//│     return (y) => {
+//│       let tmp, idx1, idx2, idx3;
+//│       idx1 = idx + 0;
+//│       tmp = buf.buf.at(idx1) + 1;
+//│       idx3 = idx + 0;
+//│       buf.buf[idx3] = tmp;
+//│       idx2 = idx + 0;
+//│       return buf.buf.at(idx2)
+//│     }
+//│   }
+//│   #x;
+//│   f(y) {
+//│     let tmp;
+//│     tmp = this.#x + 1;
+//│     this.#x = tmp;
+//│     return this.#x
+//│   }
+//│   toString() { return runtime.render(this); }
+//│   static [definitionMetadata] = ["class", "A", [null]]; 
+//│ });
+
+
+let buf = new DefaultObjectBuffer(10)
+//│ buf = DefaultObjectBuffer(_)
+
+let obj1 = buf.mkNew(A.class)(3)
+//│ obj1 = 0
+
+let obj2 = buf.mkNew(A.class)(5)
+//│ obj2 = 2
+
+A.class.f(buf, obj1)(1)
+//│ = 4
+
+A.class.f(buf, obj1)(1)
+//│ = 5
+
+A.class.f(buf, obj2)(1)
+//│ = 6
+
+buf.del(A.class, obj1)
+
+let obj3 = buf.mkNew(A.class)(10)
+//│ obj3 = 0
+
+// UAF
+A.class.f(buf, obj1)(1)
+//│ = 11
+
+A.class.f(buf, obj3)(1)
+//│ = 12
+
+
+:sjs
```

Comment:
Again, there's not need to add so much generated code to git. We already have a case like this above. Maybe consolidate both things in a single example with `:sjs`?